pkg/static/login.js

```diff
@@ -1,5 +1,10 @@
 import "./login.scss";
 
+function debug(...args) {
+    if (window.debugging == 'all' || window.debugging?.includes('login'))
```

Comment:
`===`?

Reply:
Also, this `debugging?.includes()` business is black magic :)  I'd have expected `.?()` to be needed there as well...  I guess that has to do with Javascript's strange `this` binding logic... TIL.

Reply:
Can do `===`, that was mostly copy pasta from all the other places.

I don't understand your `.?()` comment though -- either `window.debugging` is set, then it has to be a string and has an `includes()` method. Or it's undefined, then `debugging?` short-circuits.

Reply:
The issue is that I read this parsed like so:

```js

   ( (window . debugging) ?. includes ) ()

```

and assuming that `window.debugging` is undefined, then I'd also expect `(window.debugging)?.includes` to be indefined.  You can't call `undefined()`, which is why I'd have expected a `?.()` to be required.  But your code works, I tested it.  I just don't understand why.

Reply:
We have a dozen copy of that code everywhere. Indeed `?.` is a bit magic for function calls, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining.

I.e. `foo?.method(...)` is `undefined` (and nothing gets called) iff `foo` is undefined or null.